app/models/assignment.rb

```diff
@@ -1199,6 +1197,37 @@ def assignment_properties_config
     end
   end
 
+  # Writes this assignment's starter file settings to the file located at +settings_filepath+ located in the
+  # +zip_file+. Also writes the starter files for this assignment in the same directory as +settings_filepath+.
+  def starter_file_config_to_zip(zip_file, settings_filepath)
+    default_starter_group = nil
+    download_count = 1
+    group_data = []
+    directory_path = File.dirname(settings_filepath)
+    self.starter_file_groups.find_each do |starter_file_group|
```

Comment:
There's an `each_with_index` method too if you don't want to have to keep track of the counter manually.